### PR TITLE
Use fieldSelector for target shoot command

### LINKED ISSUE #127: Targeting shoots may take long
**Describe the bug**
On large landscapes with many shoots, `gardenctl target shoot` may take very long.

**To Reproduce**
See above.

**Expected behavior**
The command should finish within a second or so.

**Gardenctl Version (please complete the following information):**
```
$ g version
gardenctl:
	version     : 0.11.0
	build date  : 2019-06-14
	go version  : go1.12.5
	go compiler : gc
	platform    : darwin/amd64
```

### PR DESCRIPTION
**Pull request by ialidzhikov**
**What this PR does / why we need it**:
Use fieldSelector for `gardenctl target shoot NAME`.

**Which issue(s) this PR fixes**:
Fixes #127

**Special notes for your reviewer**:

**Release note**:

```improvement operator
`gardenctl target shoot` performance is now improved.
```

### CODE REVIEW

File: pkg/cmd/target.go [hunk 1]
```diff
@@ -413,21 +414,29 @@ func targetSeed(targetReader TargetReader, targetWriter TargetWriter, name strin
...
 		project, err := gardenClientset.CoreV1alpha1().Projects().Get(projectName, metav1.GetOptions{})
 		checkError(err)
 
 		projectNamespace := project.Spec.Namespace
-		shootList, err = gardenClientset.CoreV1alpha1().Shoots(*projectNamespace).List(metav1.ListOptions{})
+		shootList, err = gardenClientset.CoreV1alpha1().Shoots(*projectNamespace).List(listOptions)
 		checkError(err)
 	} else if len(target.Stack()) > 1 && target.Stack()[1].Kind == TargetKindSeed {
-		shootList, err = gardenClientset.CoreV1alpha1().Shoots("").List(metav1.ListOptions{})
+		shootList, err = gardenClientset.CoreV1alpha1().Shoots(metav1.NamespaceAll).List(listOptions)
```
Comment (vpnachev) [inline pkg/cmd/target.go]: in this case field selector `spec.cloud.seed=<seed-name>` can be used, for details see https://github.com/gardener/gardener/pull/677

Comment (ialidzhikov) [reply pkg/cmd/target.go]:
```
Error from server (BadRequest): Unable to find "core.gardener.cloud/v1alpha1, Resource=shoots" that match label selector "", field selector "spec.seedName=aws": "spec.seedName" is not a known field selector: only "metadata.name", "metadata.namespace"
```

Looks like this feature is gone with the new apiVersion. Will check how to fix it in g/g.

Comment (ialidzhikov) [reply pkg/cmd/target.go]: Btw can we proceed without the fieldSelector by seedName as it will take time until g/g is released and etc.?

Comment (vpnachev) [reply pkg/cmd/target.go]: yes, sure.
I was not aware that this feature was not migrated to the new API

**Pull request merged**